Client/Frontend/Browser/BrowserViewController/OpenSearch/BrowserViewController+OpenSearch.swift

```diff
@@ -0,0 +1,217 @@
+        case title = "title"
+    }
+}
+
+// MARK: - OpenSearch Browser Extension
+
+extension BrowserViewController {
+    
+    /// Adding Toolbar button over the keyboard for adding Open Search Engine
+    /// - Parameter webView: webview triggered open seach engine
+    func evaluateWebsiteSupportOpenSearchEngine(_ webView: WKWebView) {
+        let script = """
+                        var link = document.querySelector("link[type='application/opensearchdescription+xml']")
+                        var dict = {
+                            href : link.getAttribute("href"),
+                            title : link.getAttribute("title")
+                        };
+                        JSON.stringify(dict)
+                     """
+        
+        webView.evaluateJavaScript(script) { (result, _) in
+            guard let htmlStr = result as? String,
+                  let data: Data = htmlStr.data(using: .utf8) else { return }
+                        
+            do {
+                let openSearchReference = try JSONDecoder().decode(OpenSearchReference.self, from: data)
+                self.updateAddOpenSearchEngine(webView, referenceObject: openSearchReference)
+            } catch {
+                log.error(error.localizedDescription)
+            }
+        }
+    }
+    
+    private func updateAddOpenSearchEngine(_ webView: WKWebView, referenceObject: OpenSearchReference) {
+        // Add Reference Object as Open Search Engine
+        openSearchEngine = referenceObject
+        
+        // Open Search guidlines requires Title to be same as Short Name but it is not enforced,
+        // thus in case of yahoo.com the title is 'Yahoo Search' and Shortname is 'Yahoo'
+        // Instead we are checking referenceURL match to determine searchEngine is added or not
+        
+        let matches = self.profile.searchEngines.orderedEngines.filter {$0.referenceURL == referenceObject.reference}
+        
+        if !matches.isEmpty {
+            self.customSearchEngineButton.state = .disabled
+        } else {
+            self.customSearchEngineButton.state = .enabled
+        }
+        
+        /*
+         This is how we access hidden views in the WKContentView
+         Using the public headers we can find the keyboard accessoryView which is not usually available.
+         Specific values here are from the WKContentView headers.
+         https://github.com/JaviSoto/iOS9-Runtime-Headers/blob/master/Frameworks/WebKit.framework/WKContentView.h
+         */
+        guard let webContentView = UIView.findSubViewWithFirstResponder(webView) else {
+            /*
+             In some cases the URL bar can trigger the keyboard notification. In that case the webview isnt the first responder
+             and a search button should not be added.
+             */
+            return
+        }
+        
+        let argumentNextItem: [Any] = ["_n", "extI", "tem"]
+        let argumentView: [Any] = ["v", "ie", "w"]
```

Comment:
It is actually necessary? What happens if you get the first responder and then iterate upwards to the parent to check if it is actually the WKWebView? Would that work instead of using a private API?

Reply:
The problem is not to find first responder and iterate upwards. We already do that however the problem is where to put this button on the toolbar, thats why are using nextItem and constraint according to it.
Would like to try If you have a better solution.